src/hooks/use-fetch.ts

```diff
@@ -0,0 +1,47 @@
+import { useEffect, useState } from 'react'
+
+export const useFetch = <DataType>(url: string, trigger = true, requestOptions?: RequestInit) => {
+    const [error, setError] = useState<Error | undefined>(undefined)
+
+    const [data, setData] = useState<DataType | undefined>(undefined)
+
+    const [isLoading, setIsLoading] = useState(!trigger || false)
+
+    const [isTriggered, setIsTriggered] = useState(trigger)
+
+    const startTrigger = () => setIsTriggered(true)
+
+    const fetchData = async () => {
+        try {
+            const response = await fetch(url, requestOptions)
```

Comment:
I expect to change the loading state here. It should be true before starting fetching the data.

In this example :


```tsx

type Product = {
        id: string
        title: string
    }

    const { data, trigger, isLoading } = useFetch<Product[]>('https://fakestoreapi.com/products')

    useEffect(() => {
        if (data) {
            const [firstProduct, ...restProducts] = data
            console.log(firstProduct.title)
        }
    }, [data])

    useEffect(()=>{
        console.log(isLoading)
    },[isLoading])

    if (data) {
        return <p>{data[0].title}</p>
    }

    if (isLoading) {
        return <p>Loading ...</p>
    }
}
```

`isLoading` is usually `false`.

Reply:
This is a demo of the issue:

### When there is no `setLoading(true)`
[f713ab7a-1001-4b66-a068-607b43c8616f.webm](https://github.com/alamenai/rehook/assets/20143684/1a25d161-9792-4c81-8214-26d8f7b40fbf)

### When there is `setLoading(true)`

[1072e0f4-60ae-418a-acb3-7782465d477b.webm](https://github.com/alamenai/rehook/assets/20143684/6da94a4f-e3f1-44b8-bb6e-d7ed4bfbf569)

Reply:
Yes, it's fixed though this [commit.](https://github.com/alamenai/rehook/pull/43/commits/8d1e786bc4dba1e968df196c1437debd684ce1bf)